### PR TITLE
[SPARK-5966][WIP]<Spark-submit deploy-mode cluster is not compatible with master local>

### PR DESCRIPTION
**Pull request by kevinyu98**
… master local>

### CODE REVIEW

**Comment by srowen** [comment]
LGTM

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala [hunk 1]
```diff
@@ -266,6 +266,11 @@ object SparkSubmit {
       }
     }
 
+    // SPARK-5966, check deployMode CLUSTER and master local
+    if (clusterManager == LOCAL && deployMode == CLUSTER) {
+      printErrorAndExit("Cluster deploy mode is not compatible with master \"local\"")
+    }
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: there's a place down there (L321) where we do checks like this. Please move it down there to be more consistent

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/impatient.sc [hunk 2]
```diff
@@ -0,0 +1 @@
+1+1;
```
Comment (JoshRosen) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/impatient.sc]: Why this change? Unintentional?

Comment (kevinyu98) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/impatient.sc]: Hello Josh: Sorry, it is unintentional. I have deleted the new file and push again.

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala (hunk 1 above)
Comment (kevinyu98) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: Hello Andrew: Thanks for pointing this out, I have made the code changes, run the test , and submit the pull request, can you help review? 
Kevin

**Comment by andrewor14** [comment]
retest this please

**Comment by andrewor14** [comment]
LGTM, will merge once tests pass.

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala [hunk 3]
```diff
@@ -266,6 +266,7 @@ object SparkSubmit {
       }
     }
 
+
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: nit: I'm going to nix this blank line when I merge (no action required on your part)

**Comment by srowen** [comment]
Merged to master/1.5 (after removing that blank line)
